public/js/controllers/charts_controller.js

```diff
@@ -379,24 +415,15 @@ export default class extends Controller {
     }
   }
 
-  onZoom (event) {
-    var target = event.srcElement || event.target
-    this.zoomOptionTargets.forEach((el) => {
-      el.classList.remove('active')
-    })
-    target.classList.add('active')
-    this.settings.zoom = null
-    this.validateZoom()
-  }
-
   async validateZoom () {
     await animationFrame()
     this.chartWrapperTarget.classList.add('loading')
     await animationFrame()
     let oldLimits = this.limits || this.chartsView.xAxisExtremes()
     this.limits = this.chartsView.xAxisExtremes()
-    if (this.selectedZoom) {
-      this.lastZoom = Zoom.validate(this.selectedZoom, this.limits, blockTime)
+    var selected = this.selectedZoom()
+    if (selected) {
+      this.lastZoom = Zoom.validate(selected, this.limits, avgBlockTime, this.isTimeAxis())
```

Comment:
Leave `Zoom.validate` the way it was, and use the `minSize` and `scale` parameters like this

```
this.lastZoom = Zoom.validate(selected, this.limits, this.isTimeAxis() ? avgBlockTime : 1, this.isTimeAxis() ? 1 : avgBlockTime)
```